egs/wsj/s5/steps/libs/nnet3/train/common.py

```diff
@@ -318,6 +318,222 @@ def get_learning_rate(iter, num_jobs, num_iters, num_archives_processed,
     return num_jobs * effective_learning_rate
 
 
+def parse_dropout_option(num_archives_to_process, dropout_option):
+    """Parses the string option to --trainer.dropout-schedule and
+    returns a list of dropout schedules for different component name patterns.
+    Calls _parse_dropout_string() function for each component name pattern
+    in the option.
+
+    Arguments:
+        dropout_option: may have different rules for different component-name
```

Comment:
You are starting with the most complex case here and not describing the most simple, common case where there are no names.
This documentation probably requires an example.